Source/WebCore/platform/graphics/displaylists/DisplayListItems.h

```diff
@@ -950,6 +950,26 @@ class FillRectWithGradient {
     mutable Ref<Gradient> m_gradient; // FIXME: Make this not mutable
 };
 
+class FillRectWithGradientAndSpaceTransform {
+public:
+    static constexpr char name[] = "fill-rect-with-gradient-and-space-transform";
+
+    WEBCORE_EXPORT FillRectWithGradientAndSpaceTransform(const FloatRect&, Gradient&, const AffineTransform&);
+    WEBCORE_EXPORT FillRectWithGradientAndSpaceTransform(FloatRect&&, Ref<Gradient>&&, AffineTransform&&);
+
+    const FloatRect& rect() const { return m_rect; }
+    const Ref<Gradient>& gradient() const { return m_gradient; }
+    const AffineTransform& gradientSpaceTransform() const { return m_gradientSpaceTransform; }
+
+    WEBCORE_EXPORT void apply(GraphicsContext&) const;
+    void dump(TextStream&, OptionSet<AsTextFlag>) const;
+
+private:
+    FloatRect m_rect;
+    mutable Ref<Gradient> m_gradient; // FIXME: Make this not mutable
```

Comment:
Yes I think we should make the `apply` methods non const functions.

Reply:
I will have a look at it.